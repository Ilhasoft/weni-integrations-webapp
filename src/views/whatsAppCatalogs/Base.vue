<template>
  <div class="whatsapp-catalogs-base">
    <unnnic-breadcrumb :crumbs="crumbs" @crumbClick="handleCrumbClick" />
    <router-view />
  </div>
</template>

<script>
  export default {
    name: 'WhatsAppCatalogs',
    data() {
      return {
        crumb_title: this.$route.meta.crumb_title,
        baseCrumbs: [
          {
            name: this.$t('apps.nav.my_apps'),
            path: '/apps/my',
            meta: 'myApps',
          },
        ],
      };
    },
    computed: {
      crumbs() {
        const routeCrumbs = this.$route.matched.slice(1);

        const routeCrumbsWithTitles = routeCrumbs.map((crumb) => {
          const { meta } = crumb;
          return {
            name: this.$t(meta.crumb_title),
            path: crumb.path,
            meta: crumb.name,
          };
        });
        return this.baseCrumbs.concat(routeCrumbsWithTitles);
      },
    },
    methods: {
      handleCrumbClick(crumb) {
        if (crumb.meta === this.$route.name) return;

        if (crumb.meta === 'WhatsApp Catalogs List') {
          this.$router.go(-1);
        } else {
          this.$router.push(crumb.path);
        }
      },
    },
  };
</script>

<style lang="scss" scoped>
  .whatsapp-catalogs-base {
    display: flex;
    flex-direction: column;
    height: 100%;
    overflow: hidden;
  }
</style>
