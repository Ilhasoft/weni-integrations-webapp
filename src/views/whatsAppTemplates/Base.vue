<template>
  <div class="whatsapp-templates-base">
    <unnnic-breadcrumb :crumbs="crumbs" @crumbClick="handleCrumbClick" />
    <router-view />
  </div>
</template>

<script>
  export default {
    name: 'WhatsAppTemplates',
    data() {
      return {
        crumb_title: this.$route.meta.crumb_title,
      };
    },
    computed: {
      crumbs() {
        return [
          {
            name: 'Templates',
            path: '/apps/my',
          },
          {
            name: this.$t(this.crumb_title),
          },
        ];
      },
    },
    watch: {
      '$route.meta.crumb_title'(newValue) {
        this.crumb_title = newValue;
      },
    },
    methods: {
      handleCrumbClick(crumb) {
        if (crumb.path) {
          this.$router.go(-1);
        }
      },
    },
  };
</script>

<style lang="scss" scoped>
  .whatsapp-templates-base {
    height: 100%;
  }
</style>

<style lang="scss">
  body {
    margin: 0;
    padding: $unnnic-spacing-inset-md;
  }
</style>
