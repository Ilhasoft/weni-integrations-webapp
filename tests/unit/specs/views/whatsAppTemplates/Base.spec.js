import { shallowMount, createLocalVue } from '@vue/test-utils';
import Base from '@/views/whatsAppTemplates/Base.vue';

const localVue = createLocalVue();

describe('views/whatsAppTemplates/Base.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Base, {
      localVue,
      stubs: {
        UnnnicBreadcrumb: true,
        routerView: true,
      },
      mocks: {
        $t: () => 'some specific text',
        $router: {
          push: jest.fn(),
          go: jest.fn(),
        },
        $route: {
          meta: {
            crumb_title: 'title',
          },
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('watchers', () => {
    it('should change crumb_title on $route.meta.crumb_title change', async () => {
      const newTitle = 'new_title';
      expect(wrapper.vm.crumb_title).not.toEqual(newTitle);
      wrapper.vm.$route.meta.crumb_title = newTitle;
      await wrapper.vm.$nextTick();
      expect(wrapper.vm.crumb_title).toEqual(newTitle);
    });
  });

  describe('handleCrumbClick()', () => {
    it('should call router.go(-1) path is defined', () => {
      const spy = spyOn(wrapper.vm.$router, 'go');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.handleCrumbClick({ path: '/path' });
      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith(-1);
    });

    it('should not call router.go if path is not defined', () => {
      const spy = spyOn(wrapper.vm.$router, 'go');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.handleCrumbClick({});
      expect(spy).not.toHaveBeenCalled();
    });
  });
});
