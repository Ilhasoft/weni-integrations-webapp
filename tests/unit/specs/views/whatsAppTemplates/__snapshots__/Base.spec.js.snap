// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/whatsAppTemplates/Base.vue should be rendered properly 1`] = `
<div class="whatsapp-templates-base">
  <unnnicbreadcrumb-stub crumbs="[object Object],[object Object]"></unnnicbreadcrumb-stub>
  <routerview-stub></routerview-stub>
</div>
`;
